### ISSUE TITLE
margin not applied inside HStack

### ISSUE THREAD

**Issue by villesau** [chakra-ui/chakra-ui#7098]
### Description

`marginLeft` in a component inside `HStack` is not applied.

### Link to Reproduction

https://codesandbox.io/s/confident-dawn-rc5gsz?file=/src/index.tsx

### Steps to reproduce

1. Go to devtools
2. See the margin applied to the second "Box"
3. disabling the `margin-left` does not have any effect, it's not applied

### Chakra UI Version

2.4.3

### Browser

108.0.5359.98

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by bharat-dussa** [comment]
yes, same issue happened when i worked with this in one of my project

**Comment by villesau** [comment]
@segunadebayo Ok, thanks! It's such a surprising feature (?) that would likely make sense to document it somewhere. I couldn't find any mention of such behaviour in HStack docs.
